File: backend/app/schemas/knowledge.py
from pydantic import BaseModel
from datetime import datetime


class ThoughtChainCreate(BaseModel):
    workspace_id: int
    problem: str
    thought_chain: list[dict]
    tags: list[str] = []
    source: str = "manual"
    related_issue_ids: list[int] = []


class ThoughtChainResponse(BaseModel):
    id: int
    workspace_id: int
    problem: str
    thought_chain: list[dict]
    tags: list[str]
    source: str
    related_issue_ids: list[int]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class KnowledgeExtractRequest(BaseModel):
    conversation_id: int

File: frontend/src/types/knowledge.ts
export interface ThoughtStep {
  step?: number
  content: string
  role?: string
}

export interface ThoughtChain {
  id: number
  workspace_id: number
  problem: string
  thought_chain: ThoughtStep[]
  tags: string[]
  source: 'auto_extract' | 'manual'
  related_issue_ids: number[]
  created_at: string
  updated_at: string
}

export interface ThoughtChainCreate {
  workspace_id: number
  problem: string
  thought_chain: ThoughtStep[]
  tags?: string[]
  source?: 'auto_extract' | 'manual'
  related_issue_ids?: number[]
}
